src/docfx/restore/RestoreGit.cs

```diff
@@ -42,162 +35,80 @@ public static async Task<IEnumerable<(string href, string workTreeHead)>> Restor
+                var workTreeHeads = await RestoreWorkTree.AddWorkTrees(restoreDir, hrefs, token);
 
                 foreach (var (_, workTreeHead) in workTreeHeads)
                 {
-                    var childDir = GetRestoreWorkTreeDir(restoreDir, workTreeHead);
+                    var childDir = RestoreWorkTree.GetRestoreWorkTreeDir(restoreDir, workTreeHead);
                     await restoreChild(childDir);
                 }
 
                 return workTreeHeads;
             }
-        }
-
-        public static async Task GC(Config config, Func<string, Task> gcChild)
-        {
-            var restoreDirs = config.Dependencies.Values.GroupBy(d => GetRestoreRootDir(d), PathUtility.PathComparer).Select(g => g.Key);
-
-            await ParallelUtility.ForEach(
-               restoreDirs,
-               async restoreDir =>
-               {
-                   var leftWorkTrees = await CleanupWorkTrees(restoreDir);
-                   foreach (var leftWorkTree in leftWorkTrees)
-                   {
-                       await gcChild(leftWorkTree);
-                   }
-               },
-               progress: Progress.Update);
-        }
 
-        // Restore dependent repo to local and create work tree
-        private static async Task<List<(string href, string head)>> AddWorkTrees(string restoreDir, List<string> hrefs, string token)
-        {
-            var restorePath = PathUtility.NormalizeFolder(Path.Combine(restoreDir, ".git"));
-            var (url, _) = GitUtility.GetGitRemoteInfo(hrefs.First());
-            var workTreeHeads = new ConcurrentBag<(string href, string head)>();
-
-            await ProcessUtility.RunInsideMutex(
-                PathUtility.NormalizeFile(Path.GetRelativePath(AppData.GitRestoreDir, restorePath)),
-                async () =>
-                {
-                    try
-                    {
-                        await FetchOrCloneRepo();
-                        await AddWorkTrees();
-                    }
-                    catch (InvalidOperationException ex)
-                    {
-                        throw Errors.GitCloneFailed(hrefs.First()).ToException(ex);
-                    }
-                });
-
-            Task FetchOrCloneRepo()
+            List<(string restoreDir, List<string> hrefs)> GetRestoreItems()
             {
-                if (GitUtility.IsRepo(restoreDir))
-                {
-                    // already exists, just pull the new updates from remote
-                    // fetch bare repo: https://stackoverflow.com/questions/3382679/how-do-i-update-my-bare-repo
-                    return GitUtility.Fetch(restorePath, url, "+refs/heads/*:refs/heads/*", token);
-                }
-                else
+                // restore dependency repositories
+                var restoreItems = config.Dependencies.Values.GroupBy(d => GetRestoreRootDir(d), PathUtility.PathComparer).Select(g => (g.Key, g.Distinct().ToList())).ToList();
```

Comment:
can we have a method that returns all git dependencies: `IEnumerable<(string? name, string remote)> GetGitDependencies(Config config, string locale)`

Reply:
already have, RestoreGit.GetRestoreItems()